Switch to Reference Design Template codebase

MarkoPura opened the pull request:
Golioth OpenThread Demo based on the Reference Design Template with support for nRF52840 DK and Adafruit nRF52840 Feather commercially available development boards.

Need to create a release and update the link in README to projects.golioth.io.

## Testing

### nRF52840DK
  * [x] Shell/Credentials
  * [x] Connection
  * [x] Logs
  * [x] State
  * [x] Stream
  * [x] Button press
  * [x] Settings (Loop Delay)
  * [x] RPC (log level)
  * [x] RPC(reboot)
  * [x] OTA Update

### Adafruit Feather 52840
  * [x] Shell/Credentials
  * [x] Connection
  * [x] Logs
  * [x] State
  * [x] Stream
  * [x] Button press
  * [x] Settings (Loop Delay)
  * [x] RPC (log level)
  * [x] RPC(reboot)
  * [x] OTA Update

szczys committed:
Initial commit

Run the hello app by after setting credentials in the shell.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add code of conduct

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
Non-working aludel definition

mniestroj committed:
zephyr: add module.yml with configuration for boards/ directory

Defining `buiid/setings/board_root` makes Zephyr build system aware of
downstream boards defined in `boards/` directory.

mniestroj committed:
boards: aludel_mini_v1_sparkfun9160: fix copy-pasted model and compat

Those were copy-pasted from another board, so fix it now.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>

mniestroj committed:
boards: aludel_mini_v1_sparkfun9160: enable BUILD_WITH_TFM by default

Just make sure that this board definition follows other nRF9160 boards
default settings.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>

szczys committed:
dts: map golioth led, user button, click i2c

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
golioth-boards: include definitions in manifest

This commit updates the manifest file to use a separate repository for
Zephyr board definitions of custom Golioth boards. The repo will clone
into app/golioth-boards and the project is now configured to use that
directory by default for the BOARD_ROOT.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
remove credentials example

Credentials are set via the serial terminal so there is no need for a
separate credentials file.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add support for Golioth logo LED on connect

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add OTA/DFU

* Pull Golioth up to tip of main to incorporate latest OTA fixes
* Add DFU functionality

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
make DFU non-blocking

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
abstract DFU into separate file

Move all DFU-related code to app_dfu.c

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
readme: fix formatting and add settings service

* Fix headings
* Add instrcutions about LOOP_DELAY_S setting

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add user button

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add settings service

Settings service is used to adjust the LOOP_DELAY_S which is used for
the main loop sleep. When a new setting is received, the main loop
wakes, immediately taking a reading, then uses new value for sleep. The
user button will now also wake the main loop.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add app_work

Use the app_work.c file to run all of your specialized functions and
keep main and other template files unchanged.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
settings: add changed value check; increase limit

* Only perform update if received value is different from local value
* Increase LOOP_DELAY_S to 12 hours (was 100 seconds)

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
dfu: move to dfu folder

DFU uses multiple c/h files and will be easier to understand if they are
in their own folder. Rename app_dfu to golioth_dfu for clarity.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
settings: move to app_settings.c

* Move settings to its own file
* Update settings changed-value check to be more readable

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
rpc: add remote procedure call

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
settings: add register settings function

Registering a settings callback is now handles in app_settings.c by
calling app_register_settings(client); from on_connect in main.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
rpc: use RPC to reboot device

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
refactor project name to rd_template

Use `template` or `rd_template` as a generic name for the template
project.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
remove sparkfun board conf which is not up-to-date

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add workaround for duplicate observations

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
update Golioth SDK to v0.5.0

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add license

szczys committed:
fix: duplicate observations workaround

The duplicate observations workaround was previously committed without
an incrementing iteration count which caused obersavtions to never be
registered.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add device state

This commit adds device state to the template. It will observe a
`desired` endpoint and report to a `status` endpoint. (Both endpoints
are configurable in app_state.h).

The approach uses the JSON library for parsing, and tests two values
`example_int0` and `example_int1` for bounding [0..9999].

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Fix some typos in the README

szczys committed:
add set_log_level RPC

Sending a value [0..4] to the set_log_level RPC will reset the
configured log level for all logging sources. This is a runtime setting
and will revert to the compiled-in log level after the next power cycle.

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Fix title underline too short warning

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
remove work queue from app_work

The main loop not directly calls the sensor reading function.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
remove work queue for updating state

* State update already happens with async call to Golioth LightDB state
  so a worker is not needed.
* Use semaphore to ensure state update happens when first connected but
  not on reconnects.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
remove work queue for setting default endpoint

The "desired" endpoint can be set to defaults using an async call to the
Golioth LightDB State. A worker is not needed for this process.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
update Golioth SDK for observation fix

This PR updates the Golioth SDK to include a fix for the
double-observation issue. The workaround has been removed from
on_connect()

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add board files for the nrf9160dk

IMPORTANT: The i2c and UART lines have been mapped in the overlay file
but they have not been tested.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add missing void in function declaration

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
better naming for sensor_read

rename app_work_sensor_reading() to app_work_sensor_read()

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
update golioth-zephyr-boards submodule hash

Includes new pinctrl assignments for RX/TX

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add manual control for LTE connection

Manually control the LTE link so that we can manage screen/LED updates
at power-up before LTE connection begins.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add non-blocking connection option

When CONFIG_LTE_AUTO_INIT_AND_CONNECT is `n` the network and Golioth
will be manually started in a non-blocking way. This will allow sensor
readings to start before the nework is avaialble.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
show counters in Ostentus slideshow

One slide displays an upcounting number, the other a down-counting
number.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
use enum in stead of magic numbers for slide key

Ostentus slides each need their own unique key for sending value
updates. This commit moves from using magic numbers to using an enum to
specify those keys.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add Ostentus

* add libostentus to west manifest
* add show Golioth splash at power-up
* turn on power icon at power-up
* turn on internet icon when LTE connects
* turn on Golioth icon when Golioth connects

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add example of custom summary slide name

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
fix typo in comment

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Add descriptions for settings & RPCs

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
workaround for reporting DFU status

When manually controlling the LTE Link, the DFU status update cannot be
called from on_connect() because it is synchronous. This commit adds a
semaphore that is given the first time through on_connect() and taken by
the main() loop (without blocking) to report DFU status to Golioth.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
update golioth-zephyr-boards to fix reset

pulls in latest commit which changes runner order so that the board will
automatically reset after flashing with a J-Link programmer.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add log message to display firmware version

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
turn on battery icon with the power icon

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add checkpatch.conf and adhere to it

This commit brings code style up to the standards used across all
Golioth firmware projects.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
lock libostentus to a specific hash

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
add battery level logging

Add the voltage divider on the Sparkfun Things Plug nRF9160 to read the
battery voltage and estimate charge level.

based on https://github.com/golioth/reference-design-air-quality/commit/21d0fd5f28ec488810757af44d3856c93f0a7de3

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
implement a battery_monitor for Aludel-mini

This commit moves the battery monitoring files into their own directory
and adds the `CONFIG_ALUDEL_BATTERY_MONITOR` Kconfig symbol that will
include them in the build.

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Add a function to read the battery info and level

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Fix nRF9160-DK overlay

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Change template_* variables to app_*

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Update Golioth Zephyr SDK to v0.6.0

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Add settings debug messages

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Merge pull request #22 from golioth/cdwilson/upgrade-sdk-to-v0.6.0

Update Golioth Zephyr SDK to v0.6.0

cdwilson committed:
Change golioth-led to led2 on nRF9160-DK board

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
battery_monitor: fix deprecated header include

Include zephyr/kernel.h instead of zephyr/zephyr.h

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Merge pull request #23 from golioth/cdwilson/nrf9160-dk-golioth-led

Change golioth-led to led2 on nRF9160-DK board

cdwilson committed:
Merge pull request #24 from golioth/szczys/fix-deprecated-header

battery_monitor: fix deprecated header include

cdwilson committed:
Fix path to typedefsfile

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Merge pull request #25 from golioth/cdwilson/fix-typedefsfile

cdwilson committed:
Add missing include in battery header

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Fix battery measurement log message

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Add documentation for the nRF9160-DK to the README

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Use Arduino header DT node labels

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Fix battery log messages

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Add battery and firmware slides

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
README: add information about extneral libraries

Document that the `golioth-boards` and `libostentus` libraries are
included but only apply to specific custom hardware.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_rpc: flush logs after countdown

Calling `LOG_PANIC()` should be done after the countdown to ensure logs
are flushed before rebooting.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
main: document button ISR

Add warning for function called by the button interrupt

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_settings: add stdint for completeness

Include `stdint.h` for `uint32_t` style variables

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
dfu: replace dfu status workaround

Use Golioth Zephyr SDK async firmware status function which replaces
previous workaround for deadlock issue when reporting firmware status
from the on_connect callback.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_settings: normalize naming scheme

Change app_register_settings to app_settings_register

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_rpc: normalize naming scheme

Change app_register_rpc to app_rpc_register

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_rpc: return error values when available

functions that previously tested for errors but returned void were
changed to return the error values

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_settings: return error values when available

functions that previously tested for errors but returned void were
changed to return the error values

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_state: return error codes when processing

Return any error codes when processing the desired state. Previously,
there was a possibility that error codes would be generated but not
returned. This commit returns codes where possible.

Note that when updating the desired and actual endpoints, it is possible
that both will generate an error code but only the code from the desired
state operation will be returned. This is a known limitation to make
sure both operations are attempted.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_state: return error code from observe

Error codes that occur while setting up the state observe callback and
updating the initial state are now returned.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
headers: include Golioth system client

Include `net/golioth/system_client.h` in header files that declare
functions with `golioth_client` structs as paramemters.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
headers: add Doxygen description

Add a description of each service in the relevant header file. Mention
the existence of these comment blocks in the README.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
rpc: add network info

Add the network info library and call it using the `get_network_info`
RPC.

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Remove unused sensor.h include

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Select CONFIG_SENSOR when ALUDEL_BATTERY_MONITOR is enabled

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Add clang-format config

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Add editorconfig config

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Separate battery logic from app

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
battery: improve modularity of readings and slides

* Make string representation of voltage and level available using
  getters
* Gate Ostentus slides related to battery
* Use getters to supply values to Ostentus slides

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
log modem firmware version

If the MODEM_INFO library is enabled, log the modem firmware version at
power-on.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
release: v1.0.0

This commit represents the first feature-complete state of the Golioth
Reference Design Template.

Track changes using CHANGELOG.md

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
bump golioth-zephyr-boards to tip of main

Use the most recent commit for the golioth-zephyr-boards repository.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
bump zephyr-network-info to tip of main

Use newest commit of the zephyr-network-info library

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_state: fix state update and desired reset

* Properly reset desired state when example_int1 update is received but
  example_int0 is not
* Increase buffer size to accomodate 5 digit string representation of large
  uint32 numbers
* Document LightDB state behavior

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
main: fix code formatting

Follow code formtting style guidelines.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
fix code formatting

Adhere to code formatting guidelines

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
fix: turn on Golioth LED when connected

Function call to turn on Golioth LED added to on_connect()

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Spelling and formatting fixes

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Add missing license info

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Fix inline code formatting in README

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Use sizeof instead of magic numbers

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Battery code doesn't depend on zephyr sensor code

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Format source code with clang-format

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
release template_v1.0.1

See CHANGELOG for details

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_rpc: workaround set_log_level lockup

The set_log_level RPC now prints out one line recapping the number of
modules reset and the level they were reset to.

This replaces a log message for each item module, when the RPC was called
multiple times, caused deadlock behavior in the app.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
README: fix formatting

Correct the list and code block formatting in the "Using this template..."
section.

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Remove unused config symbols that cause warnings

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Fix ReST formatting and typos

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Fix name of actual state endpoint in README

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
update to Golioth v0.7.1

Update to most recent Golioth Zephyr SDK release v0.7.1 which uses:
* nRF Connect SDK v2.4.1 (NCS)
* Zephyr v3.3.99-ncs1-1

Changes in this commit:

* Updates for Golioth Zephy SDK:
  * update to use zcbor instead of qcbor
  * update RPC functions to use zcbor
  * update network-info repo revision to use new version that is zcbor
    compatible
  * update flash.c/flash.h files to match what is present in the DFU sample
  * boards/nrf9160dk_nrf9160_ns: update configuration
  * boards/aludel_mini_v1_sparkfun9160_ns: update configuration
* Updates for new NCS/Zephyr version:
  * update LTE link control: Disable samples/common link control and use
    in-app link control to start connection asynchronously
  * main.c: return int from main()
  * battery_monitor: use void as initialization param
* Misc updates:
  * golioth-boards: include as a module
  * remove module.yml: no longer needed for golioth-boards
  * network-info: include as a module
  * main: use LOG instead of printk for button errors

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
release v1.1.0

See CHANGELOG for details

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
libostentus: update to use as a Zephyr module

* Bump to libostentus v1.0.0
* Include using CONFIG_LIB_OSTENTUS
* Guard all calls to this library using the above symbol

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
west.yml: use https instead of ssh

Use publicly available https for cloning dependencies:

* golioth-zephyr-boards
* libostentus

Signed-off-by: Mike Szczys <mike@golioth.io>

MarkoPura committed:
Minor cosmetic changes
* change year to 202X-2023 in Copyright Golioth
* header file comments placed at the top for better readability

MarkoPura committed:
Clean Ostentus memory if enabled

cdwilson committed:
Upgrade golioth-zephyr-boards dependency to v1.0.1

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Merge pull request #54 from golioth/cdwilson/upgrade-golioth-zephyr-boards

Upgrade golioth-zephyr-boards dependency to v1.0.1

beriberikix committed:
Prebuilt binaries using GitHub Actions (#52)

Adds GitHub Action to build binaries for testing and release prep. Triggered manually with optional MCUboot version as an input.

beriberikix committed:
Build as a basic regression test (#56)

beriberikix committed:
Moved to nested containers and fixed with a bunch of hackery.

cdwilson committed:
Ensure Ostentus is enabled before setting slides

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
battery_monitor: only stream if connected

Check for an active connection to Golioth before trying to stream battery
readings from read_and_report_battery()

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
README: document zephyr-network-info library

The network info code was recently converted to a module and pulled in as a
library by west.yml. This commit documents its addition as an external
library.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
manifest: use Golioth Firmware SDK

Use the Golioth Firmware SDK instead of the Golioth Zephyr SDK. This
implements Golioth Firmware SDK v0.10.0

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
settings: use Golioth Firmware SDK

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
state: use Golioth Firmware SDK

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
battery_monitor: use Golioth Firmware SDK

* Pass a golioth_client_t when calling a function that streams data to
  Golioth. Clients are not singletons and this preserves the ability to use
  multiple clients.
* Use an asynchronous Stream call instead of synchronous so that we avoid
  blocking

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_work: use Golioth Firmware SDK

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_rpc: use Golioth Firmware SDK

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
boards: remove deprecated RPC length symbol

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
DFU: use Golioth Firmware SDK

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
README: update to use beta branch

This will need to be changed to remove the branch and admonition when this
is merged into main.

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Initial support for Aludel Elixir v1 board

Signed-off-by: Chris Wilson <chris@cgnd.dev>

pasrom committed:
main: Fix typo in Kconfig symbol for Golioth sample

This commit fixes a typo in the Kconfig symbol for the Golioth sample in the main.c file. The symbol should be "CONFIG_GOLIOTH_SAMPLE_COMMON" instead of "CONFIG_GOLIOTH_SAMPLES_COMMON".

cdwilson committed:
Add copyright & license info to GHA config (#63)

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
upgrade to Golioth Zephyr SDK v0.8.0

* Update Kconfig Symbols
* Use CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION for DFU version
* Remove emergency LTE state

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
release template_v1.2.0

See CHANGELOG for details

Signed-off-by: Mike Szczys <mike@golioth.io>

beriberikix committed:
Use public Docker Hub image (#67)

beriberikix committed:
Run build test on PR and Push (#68)

* Use public Docker Hub image

* Add test job

* Needs runs-on

* Proper syntax for inputs

* Checkout before calling workflow

* Reference org/repo

* Reference locally instead

* Move to job

* env not working

* Artifact conditional logic

beriberikix committed:
Update release filenames (#69)

* Use public Docker Hub image

* Add test job

* Needs runs-on

* Proper syntax for inputs

* Checkout before calling workflow

* Reference org/repo

* Reference locally instead

* Move to job

* env not working

* Artifact conditional logic

* Add tag and remove SHA from filenames

* Use Release version as tag

* Echo works this way

* $GITHUB_ENV syntax

* $GITHUB_ENV isn't working anymore for some reason.

* Stage stop not needed

* Set tag as draft title

* Match filenames to requested structure

cdwilson committed:
Ensure west init uses the correct manifest revision in GHA

Signed-off-by: Chris Wilson <chris@cgnd.dev>

MarkoPura committed:
Merge remote RD Template

* Switch codebase to Reference Design Template

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
Update config files

* Update config files to be inline with Zephyr SDK v0.8.0
* Add support for nRF52840 DK
* Add support for Adafruit nRF52840 Feather
* Remove Nordic configuration files

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
Remove battery monitor

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
west.yml: add Thread module and remove unused ones

* Remove golioth-zephyr-boards
* Remove libostentus
* Remove zephyr-network-info

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
Update CMakeLists.txt

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
app_rpc: remove get_network_info

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
app_work: remove ostentus and battery monitor code

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
main: adjust for Thread

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
Update sample.yaml

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
feat: add child_image to enable OTA updates

* Add child image for nRF52840 DK
* Add child image for Adafurit nRF52840 Feather

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
Update CHANGELOG

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
Update README

Signed-off-by: Marko Puric <marko@golioth.io>

szczys committed:
Register LTE Link Handler for connect event

Utilize the LTE Link Handler in the Golioth commmon library to log LTE
events, but register a second handler in the app to call funtions in the
case of a connect event.

This cleans up a legacy LTE Handler that reimplemented what's available in
the Golioth common library.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
use ifdef for conditional LTE code

The if_enabled syntax guarding LTE code was difficult to read. Using ifdef
makes it easier for the eye to parse.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
workflow: remove version string from build command

The firmare version is now stored in the prj.conf file. It can still be
overridden using the commandline flag:

```
west build -b <board_name> app -- -DCONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION=\"0.0.0\"j
```

However, it's better to compile in the version number that has been
committed.

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Clean up nRF9160 DK board definitions

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
main: remove redundant Golioth LED change

The Golioth LED will be turned on by the on_client_event() callback and
doesn't need to be changed after taking the semaphore.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
release template_v2.0.0

This release migrates to the Golioth Firmware SDK at v0.10.0

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
state: initialize err var to zero

This err variable must be initializated to 0 to avoid triggering an error
message when there has been no error.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
properly size stack values

The system work queue stack size is set in the prj.conf, but then half of
that size is selected in both of the board conf files. Even with the
reduced size the utilization is still only around 60% so this value is now
used in prj.conf and has been removed from board files.

The main stack size is also just over 1k so the value has been reduced from
4k to 2k in this commit.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
app_work: rename to app_sensors

The app_work.c/h files are mainly used for reading sensor data and sending
on Golioth Stream. Changing the name to app_sensors.c/h will help users
understand the purpose of the file.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
settings: wake main thread after updating delay

When the LOOP_DELAY_S setting is updated, we need to wake the main thread
which is currently sleeping based on the old delay value. This means you
will have a sensor reading every time a new setting is received, but the
point is for new setting to immediately take effect.

Signed-off-by: Mike Szczys <mike@golioth.io>

cdwilson committed:
Remove nRF DK common dtsi

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Fix typos in app_sensors_read_and_stream

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Remove remaining references to work in app_sensors

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Change app_sensors_init to app_sensors_set_client

In some of the reference designs, sensor initialization was being done
in app_sensors_init (this function also set the Golioth client for
streaming sensor data to Golioth). This function was called in main
before app_work_sensor_read, so the sensor initialization always
happened before trying to read from the sensors and send the data to
Golioth.

However, in template_v2.0.0 the call to app_sensors_init was moved into
the start_golioth_client function, which doesn't get called until the
lte_handler callback is run for cellular devices. As a result, on
nRF9160 boards app_sensors_read_and_steam ends up running before
app_sensors_init (i.e. it tries to read from the sensors before they are
initialized).

This change renames app_sensors_init to app_sensors_set_client to make
it clear that the purpose of this function is to set the Golioth client
used for streaming sensor data. Any sensor initialization should moved
to a separate function (e.g. app_sensors_init).

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Fix RPC registration error handling

Previously it was possible for some of the RPC registrations to fail,
but as long as the last one succeeded, app_rpc_register would
incorrectly return success. Since the return value of app_rpc_register
is not being checked anywhere, this just removes the return value (an
error will still be printed when any of the registrations fail).

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Remove 'v1' version specifier from Aludel Elixir board

Signed-off-by: Chris Wilson <chris@cgnd.dev>

cdwilson committed:
Upgrade to Golioth Firmware SDK at v0.11.0

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
update Golioth to v0.11.1

This incorporates a new symbol for setting the return length of an RPC.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
rpc: increase max response length

The network info module (called by get_network_info RPC) returns a large
amount of data about cellular or wifi network. Increase the RPC return
length so that data is not truncated.

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
network info: update version to newest release v1.1.1

Signed-off-by: Mike Szczys <mike@golioth.io>

mniestroj committed:
configure SDK Kconfig dependencies explicitly

Do not rely on default values pulled by CONFIG_GOLIOTH_SAMPLE_COMMON=y
option, since that is subject to west manifest ordering (or Zephyr modules
list order when west is not used). Default Kconfig values is also dropped
in newer releases of Golioth SDK.

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>

cdwilson committed:
Disable LTE auto-connect

When LTE_CONNECTIVITY is enabled, LTE_CONNECTIVITY_AUTO_CONNECT is
enabled by default, causing the Connection Manager to automatically call
LTE connect after the network interface has been brought up at boot.

The app also calls lte_lc_init_and_connect_async(lte_handler) manually,
registering an LTE handler that creates/starts the Golioth Client.

However, if the LTE registration completes before the app registers the
LTE handler, the handler will not be called, and the Golioth Client will
not be started.

This commit disables the Connection Manager LTE auto-connect
functionality, allowing the app to call connect and register the handler
manually. This ensures that the handler will be called and the Golioth
Client is started on connect.

Signed-off-by: Chris Wilson <chris@cgnd.dev>

szczys committed:
bump Golioth Firmware SDK to v0.12.2

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
release template_v2.1.0

See CHANGELOG for details

Signed-off-by: Mike Szczys <mike@golioth.io>

szczys committed:
README: remove version flag from build commands

The firmware version is set in the `prj.conf` file. Remove the command line
flag from the nrf9160dk sample build commands to align with this approach.

Signed-off-by: Mike Szczys <mike@golioth.io>

MarkoPura committed:
Merge upstream reference-design-template

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
post merge cleanup

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
change firmware version and Thread Network Name

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
turn off connection LED when disconnected

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
update README

Signed-off-by: Marko Puric <marko@golioth.io>

szczys commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,17 @@
+    uses: ./.github/workflows/build_zephyr.yml
+    with:
+      ZEPHYR_SDK: 0.16.3
+      BOARD: aludel_mini_v1_sparkfun9160_ns
```
Should this target the 52840 for automated build tests?

```suggestion
      BOARD: nrf52840dk_nrf52840
```

szczys commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,52 @@
+        strategy:
+            matrix:
+              ZEPHYR_SDK: [0.16.3]
+              BOARD: ["nrf9160dk_nrf9160_ns","aludel_mini_v1_sparkfun9160_ns"]
```
Should this target the two boards you have listed as node hardware?

szczys commented on .github/workflows/build_zephyr.yml:
```diff
@@ -0,0 +1,82 @@
+      BOARD:
+        required: true
+        type: string
+        default: aludel_mini_v1_sparkfun9160_ns
```
Maybe this should default to the 52840dk?

szczys commented on README.rst:
```diff
@@ -62,45 +72,209 @@ by changing these lines in the ``prj.conf`` configuration file, e.g.:
+
+.. code-block:: text
+
+   $ (.venv) west build -p -b <your_zephyr_board_id> app --
```
```suggestion
   $ (.venv) west build -p -b <your_zephyr_board_id> app
```

szczys commented on README.rst:
```diff
@@ -62,45 +72,209 @@ by changing these lines in the ``prj.conf`` configuration file, e.g.:
+
+.. code-block:: text
+
+   $ (.venv) west build -p -b nrf52840dk_nrf52840 app --
```
```suggestion
   $ (.venv) west build -p -b nrf52840dk_nrf52840 app
```

szczys commented on README.rst:
```diff
@@ -1,58 +1,68 @@
+firmware images <releases_>`_ for the Golioth OpenThread Demo.
 
-This is a standalone repository that will download all required files and dependencies. This directory will take up a good amount of room, as it will contain the latest Nordic NCS in the ``deps`` repository..
+[To be updated] The full project details are available on the `Golioth Thread Demo Project Page`_.
```
```suggestion
The full project details are available on the `Golioth Thread Demo Project Page`_.
```

Maybe we should update this now as we know what the projects.golioth.io URL will be

szczys commented on README.rst:
```diff
@@ -62,45 +72,209 @@ by changing these lines in the ``prj.conf`` configuration file, e.g.:
+.. _Golioth Console: https://console.golioth.io
+.. _GL-S200 Thread Border Router: https://www.gl-inet.com/products/gl-s200/
+.. _Nordic nRF52840 DK: https://www.nordicsemi.com/Products/Development-hardware/nRF52840-DK
+.. _Golioth Thread Demo Project Page: https://golioth.github.io/golioth-openthread-demo-docs
```
```suggestion
.. _Golioth Thread Demo Project Page: https://projects.github.io//reference-designs/openthread-demo/
```

szczys commented on boards/adafruit_feather_nrf52840.conf:
```diff
@@ -0,0 +1,48 @@
+CONFIG_FLASH_MAP=y
+CONFIG_NVS=y
+
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=8192
```
This seems like a really large stack. Do we need that much memory?

MarkoPura replied:
We don't, remainders from development. Changed to `2048`

szczys commented on boards/nrf52840dk_nrf52840.conf:
```diff
@@ -0,0 +1,48 @@
+CONFIG_FLASH_MAP=y
+CONFIG_NVS=y
+
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=8192
```
Same question as above... do we need this much memory?

szczys commented on src/main.c:
```diff
@@ -1,205 +1,118 @@
+#include <zephyr/net/openthread.h>
 #include <openthread/thread.h>
+#include <zephyr/net/coap.h>
+#include <zephyr/net/socket.h>
```
nit: This might be tidier if the includes were alphabetized.

szczys reviewed (approved):
Some changes to the workflows and suggestions in the README, but all looks good in general. This is a huge chunk of work, great job!

MarkoPura committed:
workflow: update BOARD

Update BOARD in github workflows from aludel_mini_v1_sparkfun9160_ns
to nrf52840dk_nrf52840 since it is used in the Follow-Along guide.

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
main: arrange header files in alphabetical order

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
update README

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
config: decrease system workqueue stack size

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
change firmware version to 1.0.0

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
update Golioth Firmware SDK to v0.13.1

Signed-off-by: Marko Puric <marko@golioth.io>

MarkoPura committed:
app_state: update lightdb_observe function call

Update golioth_lightdb_observe_async() function call
to match Golioth Firmware SDK to v0.13.1.

Signed-off-by: Marko Puric <marko@golioth.io>
